### ISSUE TITLE
WrapperMetric does not reset the state of the wrapped metric

### ISSUE THREAD

**Issue by raverank** [Lightning-AI/torchmetrics#3122]
## 🐛 Bug

When using any `WrapperMetric` (e.g. `MetricInputTransformer`, `LambdaInputTransformer`) calls to `.reset()` are not forwarded to the wrapped metric.

If any of the metrics (also in a `MetricCollection`) in a LightningModule are `WrapperMetric`s, the values will be averaged over the whole training and not just over a single epoch.

### To Reproduce

<details>
  <summary>Code sample</summary>

```python
import torch
from torchmetrics import MeanSquaredError
from torchmetrics.wrappers import LambdaInputTransformer

# Create two simple tensors
preds_1 = torch.tensor([1.0, 2.0, 3.0])  # for first evaluation
preds_2 = 2 * preds_1                    # for second evaluation
target = torch.tensor([2.0, 4.0, 6.0])


mse_metric = MeanSquaredError()

lambda_metric = LambdaInputTransformer( # using simple identity transforms! So it should end up with the same metric values
    wrapped_metric=MeanSquaredError(),
    transform_pred=lambda p: p,
    transform_target=lambda t: t,
)


mse_metric.update(preds_1, target)
lambda_metric.update(preds_1, target)
print(f"1. MSE without transformer: {mse_metric.compute().item():5.2f}, update count: {mse_metric.update_count}")
print(f"1. MSE with transformer:    {lambda_metric.compute().item():5.2f}, update count (lambda_metric): {lambda_metric.update_count}, update count (wrapped): {lambda_metric.wrapped_metric.update_count}")
# 1. MSE without transformer:  4.67, update count: 1
# 1. MSE with transformer:     4.67, update count (lambda_metric): 0, update count (wrapped): 1

mse_metric.reset()
lambda_metric.reset()  # <-- this should reset the wrapped metric!


mse_metric.update(preds_2, target)
lambda_metric.update(preds_2, target)

print(f"2. MSE without transformer: {mse_metric.compute().item():5.2f}, update count: {mse_metric.update_count}")
print(f"2. MSE with transformer:    {lambda_metric.compute().item():5.2f}, update count (lambda_metric): {lambda_metric.update_count}, update count (wrapped): {lambda_metric.wrapped_metric.update_count}")
# 2. MSE without transformer:  0.00, update count: 1
# 2. MSE with transformer:     2.33, update count (lambda_metric): 0, update count (wrapped): 2
```

</details>

<details>
  <summary>Environment</summary>

- TorchMetrics version: 1.7.1
- Python: 3.12.3
- PyTorch Version: 2.5.1

</details>
